examples/pipelines/stabilize_images.pipe.in

```diff
@@ -22,252 +22,304 @@ process input
+    feature_detector:type                                                = ocv_SURF
 
 # The OpenCV cv::Algorithm type to use for 'detector'.
-  feature_tracker:default:feature_detector:ocv:detector:type                                   = Feature2D.SURF
-  feature_tracker:default:feature_detector:ocv:detector:Feature2D.SURF:extended                = false
-  feature_tracker:default:feature_detector:ocv:detector:Feature2D.SURF:hessianThreshold        = 5000
-  feature_tracker:default:feature_detector:ocv:detector:Feature2D.SURF:nOctaveLayers           = 3
-  feature_tracker:default:feature_detector:ocv:detector:Feature2D.SURF:nOctaves                = 4
-  feature_tracker:default:feature_detector:ocv:detector:Feature2D.SURF:upright                 = true
+    block feature_detector:ocv_SURF
+      extended           = false
+      hessian_threshold  = 5000
+      n_octave_layers    = 3
+      n_octaves          = 4
+      upright            = true
+    endblock
 
 
 # Algorithm to use for 'feature_matcher'.
 # Must be one of the following options:
 #   - homography_guided
-#   - ocv
+#   - ocv_brute_force
+#   - ocv_flann_based
+#   - fundamental_matrix_guided
 #   - viscl
-  feature_tracker:default:feature_matcher:type                                                 = homography_guided
-
+#   - vxl_constrained
+    feature_matcher:type                                                 = homography_guided
 
+    block feature_matcher:homography_guided
 # Algorithm to use for 'feature_matcher'.
 # Must be one of the following options:
 #   - homography_guided
-#   - ocv
-  feature_tracker:default:feature_matcher:homography_guided:feature_matcher:type               = ocv
-
-# The OpenCV cv::Algorithm type to use for 'matcher'.
-  feature_tracker:default:feature_matcher:homography_guided:feature_matcher:ocv:matcher:type   = DescriptorMatcher.FlannBasedMatcher
+#   - ocv_brute_force
+#   - ocv_flann_based
+#   - fundamental_matrix_guided
+#   - viscl
+#   - vxl_constrained
+      feature_matcher1:type               = ocv_flann_based
 
 # Algorithm to use for 'homography_estimator'.
 # Must be one of the following options:
 #   - ocv
 #   - vxl
-  feature_tracker:default:feature_matcher:homography_guided:homography_estimator:type          = vxl
+      homography_estimator:type          = vxl
 
 # The acceptable error distance (in pixels) between warped and measured points
 # to be considered an inlier match.
-  feature_tracker:default:feature_matcher:homography_guided:inlier_scale                       = 10
-
+      inlier_scale                       = 10
+    endblock
 
 # Algorithm to use for 'loop_closer'.
 # Must be one of the following options:
 #   - bad_frames_only
+#   - exhaustive
+#   - keyframe
 #   - multi_method
 #   - vxl_homography_guided
-  feature_tracker:default:loop_closer:type                                                     = multi_method
+    loop_closer:type                                                     = multi_method
```

Comment:
The "keyframe" method is really the only one I use now.  However, it's complexity it grows with the length of the scene and the amount of change in viewpoint from frame to frame.  It is better loop closure (getting matches over frames separated by longer time).  This is important for bundle adjustment, but may not be as important if you only care about adjacent frame matching or if speed is more important.